assets/js/plugin-check-admin.js

```diff
@@ -0,0 +1,43 @@
+( function ( data ) {
+	const checkItButton = document.getElementById( 'pc_check_it' );
+	const pluginsList   = document.getElementById( 'pc_plugins' );
+
+	checkItButton.addEventListener( 'click', (e) => {
+		e.preventDefault();
+
+		const pluginCheckData = new FormData();
+
+		// Collect the data to pass along for generating a check results.
+		pluginCheckData.append( 'action', 'plugin_check_run_checks' );
+		pluginCheckData.append( 'nonce', data.nonce );
+		pluginCheckData.append( 'plugin', pluginsList.value );
+
+		fetch(
+			ajaxurl,
+			{
+				method: 'POST',
+				credentials: 'same-origin',
+				body: pluginCheckData
+			}
+		)
+		.then(
+			( response ) => {
+				if ( ! response.ok ) {
+					throw new Error(`[${response.message}]`);
+				}
+
+				return response.json();
+			}
+		)
+		.then(
+			( data ) => {
+				console.log( data.data.message );
+			}
+		)
+		.catch(
+			( error ) => { console.log( error ); }
```

Comment:
Nit-pick: Better to use `console.error`.

Also, per my other comment, we should check if a response was still returned from the server. If yes, we can extract the first error message from it and display it (`response.data[ 0 ].message`). This needs to be optional though, as an error could also occur just because e.g. there is no network connection, in which case there won't be a response.